pkg/controller/route/routecontroller.go

```diff
@@ -50,6 +50,11 @@ func New(routes cloudprovider.Routes, kubeClient clientset.Interface, clusterNam
 }
 
 func (rc *RouteController) Run(syncPeriod time.Duration) {
+	// TODO: If we do just the full Resync every 5 minutes (default value)
+	// that means that we may wait up to 5 minutes before even starting
+	// creating a route for it. This is bad.
+	// We should have a watch on node and if we observe a new node (with CIDR?)
+	// trigger reconciliation for that node.
```

Comment:
I would file an issue and we can triage it. If it's long standing, nothing huge to worry about. One thing is that "new node" in the large cluster sense is kind of gradual, though - basically everything in the first half hour looks like a new node, so that resync period could be a little relevant. One question I had is whether the reconcile loop works like `{ reconcile(); sleep(constant); }` or if it's `{ reconcile(); sleep( to-meet-five-minutes )`. (Was a question I keep meaning to answer.)